exonum-light-client/src/main/java/com/exonum/client/HttpUrlHelper.java

```diff
@@ -48,16 +49,13 @@ static HttpUrl getFullUrl(URL host, String prefix, String relativeUrl,
   }
 
   /**
-   * Removes heading slash from the path.
-   * Useful because underlying OkHttp applies slashes when constructing paths.
+   * Normalized the given URL to the canonical form.
+   * Doesn't modify letters case in the URL.
+   * Also see {@link URI#normalize()}.
    */
-  private static String normalize(String path) {
-    requireNonNull(path);
-    if (path.startsWith("/")) {
-      return path.substring(1);
-    } else {
-      return path;
-    }
+  private static HttpUrl normalize(HttpUrl url) {
+    URI normalized = url.uri().normalize();
```

Comment:
It is no longer a naïve approach 🙃 Shall we add more tests specific to this normalization algorithm to exercise that?